chain/network/src/peer.rs

```diff
@@ -347,10 +347,14 @@ impl StreamHandler<Vec<u8>, io::Error> for Peer {
             }
         };
         match (self.peer_type, self.peer_status, peer_msg) {
+            (_, PeerStatus::Connecting, PeerMessage::HandshakeWrongGenesis) => {
+                error!(target: "network", "Trying to connect to a node with a different genesis. Check the genesis config");
```

Comment:
I think it would be helpful to say which node as well.
E.g. 
`error!(target: "network", "Trying to connect to the node {}, with a different genesis. You are probably on the wrong network (remote this node form peers) or have incorrect genesis config", self.peer_id)`